duktape/src/main/jni/javascript/JavaScriptObject.cpp

```diff
@@ -206,14 +207,48 @@ duk_ret_t JavaScriptObject::finalizer(duk_context* ctx) {
 
 namespace {
 
+/**
+ * Return a function that does:
+ *  Double d;
+ *  return Integer.valueOf(d.intValue());
+ */
+std::function<jobject(JNIEnv*, jobject)> castDoubleToIntegerFunction(JNIEnv* env) {
+  jclass integerClass = env->FindClass("java/lang/Integer");
+  const jmethodID boxInteger =
+      env->GetStaticMethodID(integerClass, "valueOf", "(I)Ljava/lang/Integer;");
+  const GlobalRef integerClassRef(env, integerClass);
+  jclass doubleClass = env->FindClass("java/lang/Double");
+  const jmethodID intValue = env->GetMethodID(doubleClass, "intValue", "()I");
+  return [boxInteger, intValue, integerClassRef](JNIEnv* jniEnv, jobject doubleValue) {
+    jobject castedToInteger;
+    if (doubleValue != nullptr) {
+      jvalue result;
+      result.i = jniEnv->CallIntMethodA(doubleValue, intValue, nullptr);
+      castedToInteger = jniEnv->CallStaticObjectMethodA(
+          static_cast<jclass>(integerClassRef.get()), boxInteger, &result);
+    } else {
+      castedToInteger = nullptr;
+    }
+    return castedToInteger;
+  };
+}
+
 JavaScriptObject::MethodBody buildMethodBody(JavaTypeMap& typeMap, JNIEnv* env, jobject method,
                                              const std::string& methodName) {
   const jclass methodClass = env->GetObjectClass(method);
 
   const jmethodID getReturnType =
       env->GetMethodID(methodClass, "getReturnType", "()Ljava/lang/Class;");
   jclass returnTypeClass = static_cast<jclass>(env->CallObjectMethod(method, getReturnType));
-  const JavaType* returnType = typeMap.getBoxed(env, static_cast<jclass>(returnTypeClass));
+  // JavaScript always gives us doubles, so we convert to Integer if needed.
```

Comment:
This seems like overkill. I'm considering forbidding int/Integer results from JS calls as a simpler solution.  Thoughts?